src/azul/indexer/document.py

```diff
@@ -396,8 +400,19 @@ def translate_fields(cls,
                 return [cls.translate_fields(val, field_types, forward=forward, path=path) for val in doc]
             else:
                 assert False, (path, type(doc))
-        elif isinstance(field_types, FieldType):
-            field_type = field_types
+        elif isinstance(field_types, (list, FieldType)):
+            if isinstance(field_types, list):
```

Comment:
The repeated type checks are annoying. Try to sculpt the control flow such that you check every case exactly once.